docs/user-guide/release_version.js

```diff
@@ -0,0 +1,61 @@
+async function getMatch(match) {
+    // Check if the match is already in the session storage
+    let version = sessionStorage.getItem(`${match[0]}`);
+    let version_tag;
+
+    if (version) {
+        // If it is, parse it and use it
+        version_tag = JSON.parse(version);
+    } else {
+        const repo = match[1];
+        const stable = match[3];
+
+        version_tag = await getReleaseLink(repo, stable);
+        sessionStorage.setItem(match[0], JSON.stringify(version_tag));
+        console.log("Fetching version tag from GitHub");
+    }
+
+    return version_tag;
+}
+
+// Function to fetch the latest release tag name from a GitHub repository
+async function getReleaseLink(repo, stable="stable") {
+    const response = await fetch(`${repo}/releases`);
+    let releases = await response.json();
+    
+    switch (stable) {
+        case "any":
+            break;
+        case "stable":
+            releases = releases.filter(release => !release.prerelease);
+            break;
+        case "prerelease":
+            releases = releases.filter(release => release.prerelease);
+            break;
+    }
+
+    return releases[0].tag_name;
+    
+}
+
+async function replacePlaceholders() {
+    // Regular expression to match the placeholders
+    // and parse the api url to get the tag name
+    const regex = /\{releaselink:repo:(.*?)(:stable:(.*?))?\}/g;
```

Comment:
Are we hardcoding :stable: here? If yes, can we get "latest" and "stable" from releaselink marker instead of hardcoding here? for example:
`{releaselink:repo:https://api.github.com/repos/kubernetes-sigs/cluster-api@latest}`
`{releaselink:repo:https://api.github.com/repos/kubernetes-sigs/cluster-api@stable}`
This will make this script more usable in the future.